src/test/java/org/prebid/server/ApplicationTest.java

```diff
@@ -1496,6 +1545,37 @@ private static String jsonFrom(String file) throws IOException {
                 ApplicationTest.class.getSimpleName() + "/" + file)));
     }
 
+    private static String cacheResponseFromRequestJson(String requestAsString, String requestCacheIdMapFile) {
+        ArrayNode responseArray = mapper.createArrayNode();
+
+        try {
+            JsonNode jsonNodeRequest = mapper.readTree(requestAsString);
```

Comment:
Probably it would be easier to deal with JSONs (both reading and construction) if they were represented with POJOs. Let's try this approach as it should produce much more readable logic.